johnsonandjohnson/Bodiless-JS
# feat(component scaffold): Component Scaffolder

## Test Instructions

- Clone and checkout branch
  ```
  git clone https://github.com/johnsonandjohnson/Bodiless-JS.git component_scaffolder && \
  cd component_scaffolder && \
  git checkout -b feat/component-scaffolder origin/feat/component-scaffolder && \
  npm run setup && npm run build
  ```
- Run cmd from package path
  ```
  cd packages/bodiless-scaffold
  ./bin/scaffold.js component
  ```
- Follow the prompt to generate component code.

Notes:
- Package path can be relative or absolute, default to './' current directory.
- Script will fail on existing files, except `base/index.ts` which can be appended.


Expected:
```
.
├── base
│   └── index.ts
├── components
│   ├── Buttons
│   │   ├── ButtonsClean.ts
│   │   ├── __tests__
│   │   │   └── Buttons.test.tsx
│   │   ├── index.static.ts
│   │   ├── index.ts
│   │   └── tokens
│   │       ├── index.ts
│   │       └── kenvueButtons.ts
│   └── Card
│       ├── CardClean.ts
│       ├── __tests__
│       │   └── Card.test.tsx
│       ├── index.static.ts
│       ├── index.ts
│       └── tokens
│           ├── index.ts
│           └── kenvueCard.ts
└── shadow
    └── @bodiless
        ├── vital-buttons
        │   └── Buttons.ts
        └── vital-card
            └── Card.ts
```
